src/weather.py

```diff
@@ -0,0 +1,29 @@
+import openmeteo_requests
+import requests_cache
+import pandas as pd
+from retry_requests import retry
+
+# Setup Open-Meteo API client with cache and retry on error
+cache_session = requests_cache.CachedSession('.cache', expire_after=3600)
+retry_session = retry(cache_session, retries=5, backoff_factor=0.2)
+openmeteo = openmeteo_requests.Client(session=retry_session)
+
+def get_weather_data(latitude: float, longitude: float):
+    url = "https://api.open-meteo.com/v1/forecast"
+    params = {
+      "latitude": latitude,
+      "longitude": longitude,
+      "current": ["temperature_2m", "relative_humidity_2m", "weather_code", "wind_speed_10m"],
+      "timezone": "auto",
+      "past_days": 92
+    }
+    responses = openmeteo.weather_api(url, params=params)
```

Comment:
The idea was that you use the requests library not the openmeteo one, but is ok hehehe

Reply:
I hacked the matrix